packages/shared/components/inputs/NftInput.svelte

```diff
@@ -1,13 +1,38 @@
 <script lang="typescript">
-    import { Modal, NftSelector, SelectorInput } from 'shared/components'
+    import { Modal, SelectorInput, Text, TextType } from 'shared/components'
+    import { truncateString } from '@core/utils'
+    import { selectedAccountNfts } from '@core/nfts'
 
     export let nftId: string = ''
     export let error: string = ''
 
     let inputElement: HTMLInputElement = undefined
     let modal: Modal = undefined
+
+    export async function validate(): Promise<void> {
+        if (!nftId) {
+            error = 'Nft is required'
+            return Promise.reject(error)
+        } else {
+            return Promise.resolve()
+        }
+    }
+
+    function onClick(_selected): void {
```

Comment:
I'm missing the parameter typing here as well.

```suggestion
    function onClick(selectedNft): void {
```